engine/robot.py

```diff
@@ -253,22 +264,25 @@ def execute_setup(self):
             self.robot_state.ekf = LocalizationEKF(mu, sigma)
 
             if (self.robot_state.radio_session.connected and gps_setup and imu_setup):
-                obstacle_avoidance = threading.Thread(target=self.track_obstacle, daemon=True)
+                obstacle_avoidance = threading.Thread(
+                    target=self.track_obstacle, daemon=True)
                 obstacle_avoidance.start()  # spawn thread to monitor obstacles
                 self.set_phase(Phase.TRAVERSE)
         else:
             self.set_phase(Phase.TRAVERSE)
 
     def execute_traversal(self, unvisited_waypoints, allowed_dist_error, base_station_loc, control_mode, time_limit,
                           roomba_radius, database):
+        self.robot_state.prev_phase = Phase.TRAVERSE
```

Comment:
shouldn't this be at the end of the function? bc we also have a attribute for the current phase and this seems to be considering the prev_phase as the current phase? maybe we can add `self.robot_state.prev_phase = self.robot_state.phase` in the `set_phase` function?

Reply:
I think we want it in the beginning so if obstacle avoidance triggers, phase becomes obstacle avoidance phase, but we want to go back to prev phase

Reply:
what does wheel radius equals 0 mean

Reply:
what?